substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -3169,8 +3232,34 @@ impl<T: Config> Pallet<T> {
 				"depositor must always have MinCreateBond stake in the pool, except for when the \
 				pool is being destroyed and the depositor is the last member",
 			);
+
+			let expected_tvl: BalanceOf<T> = BondedPools::<T>::iter()
+				.map(|(id, inner)| {
+					T::Staking::total_stake(
+						&BondedPool { id, inner: inner.clone() }.bonded_account(),
+					)
+					.unwrap_or_default()
+				})
+				.reduce(|acc, total_balance| acc + total_balance)
+				.unwrap_or_default();
+
+			ensure!(
```

Comment:
Bit worried that this is actually too slow, but this is a concern independent of this pallet and similar to https://github.com/paritytech/polkadot-sdk/issues/234